automatic/jenkins/tools/chocolateyinstall.ps1

```diff
@@ -0,0 +1,115 @@
+$ErrorActionPreference = 'Stop'
+
+$ToolsDir = Split-Path -parent $MyInvocation.MyCommand.Definition
+$JenkinsRegistryRoot = "HKLM:\SOFTWARE\Jenkins\InstalledProducts\Jenkins"
+
+# If x86 exists, that is the default installation location for the previous version of Jenkins on an x64 system
+$ProgramFiles = @($env:ProgramFiles, ${env:ProgramFiles(x86)})[[bool]${env:ProgramFiles(x86)}]
+$MsiUpgrade = -not (Test-Path $JenkinsRegistryRoot) -and
+    (Test-Path $ProgramFiles\Jenkins\jenkins.exe) -and
+    [version](Get-ItemProperty $ProgramFiles\Jenkins\jenkins.exe -ErrorAction SilentlyContinue).VersionInfo.ProductVersion -le "2.3.0"
+
+$packageArgs = @{
+    packageName   = $env:ChocolateyPackageName
+    fileType      = 'msi'
+    file          = Join-Path $ToolsDir 'jenkins.msi'
+    silentArgs  = "/qn /norestart /l*v `"$($env:TEMP)\$($env:ChocolateyPackageName).$($env:chocolateyPackageVersion).MsiInstall.log`""
+    validExitCodes= @(0, 3010, 1641)
+    softwareName  = 'Jenkins*'
+}
+
+# Handle Package Parameters
+$PackageParameters = Get-PackageParameters
+foreach ($PotentialParameter in @(
+        "INSTALLDIR"
+        "JENKINS_ROOT"
+        "PORT"
+        "JAVA_HOME"
+        "SERVICE_USERNAME"
+        "SERVICE_PASSWORD"
+    )) {
+    if ($PackageParameters.ContainsKey($PotentialParameter)) {
+        Set-Variable -Name $PotentialParameter -Value $PackageParameters[$PotentialParameter]
+    }
+}
+
+if ($INSTALLDIR) {
+    $packageArgs.silentArgs += " INSTALLDIR=`"$($INSTALLDIR)`""
+}
+
+if ($JENKINS_ROOT) {
+    $packageArgs.silentArgs += " JENKINS_ROOT=`"$($JENKINS_ROOT)`""
```

Comment:
Do we really want to pass in the jenkins root even if the installed version is specifying something different?

Reply:
From memory but unconfirmed: I seem to remember that Jenkins installer doesn't do anything with it if you pass it in, at this point. (See [here](https://github.com/JPRuskin/jenkinssci-packaging/blob/master/msi/build/jenkins.wxs#L123))

Fair enough, though, we shouldn't pass it even if something won't be done with it.